src/main/java/smile/identity/core/WebApi.java

```diff
@@ -79,99 +77,111 @@ public String submit_job(String partner_params, String images_params, String id_
-            } else {
-                idInfo = (JSONObject) parser.parse(id_info_params);
-            }
-
-            Long job_type = (Long) partnerParams.get("job_type");
-            if (job_type == 5) {
-                response = callIDApi(partnerParams, idInfo);
-            } else {
-                throw new IllegalArgumentException("You need to send through more parameters");
-            }
+    	String response = null;
+        
+    	JSONParser parser = new JSONParser();
+        JSONObject partnerParams = (JSONObject) parser.parse(partner_params);
+        JSONObject idInfo;
+        
+        if (id_info_params == null && id_info_params.trim().isEmpty()) {
+            throw new IllegalArgumentException("You need to send through a vaild ID Information payload");
+        } else {
+            idInfo = (JSONObject) parser.parse(id_info_params);
+        }
 
-        } catch (Exception e) {
-            throw e;
+        Long job_type = (Long) partnerParams.get("job_type");
+        
+        if (job_type == 5) {
+            response = callIDApi(partnerParams, idInfo, null);
+        } else {
+            throw new IllegalArgumentException("You need to send through more parameters");
         }
+        
         return response;
     }
 
     public String get_job_status(String partner_params, String options) throws Exception {
-        String response = null;
-
-        try {
-            JSONParser parser = new JSONParser();
-            JSONObject partnerParams = (JSONObject) parser.parse(partner_params);
+        JSONParser parser = new JSONParser();
+        JSONObject partnerParams = (JSONObject) parser.parse(partner_params);
 
-            String user_id = (String) partnerParams.get("user_id");
-            String job_id = (String) partnerParams.get("job_id");
+        String user_id = (String) partnerParams.get("user_id");
+        String job_id = (String) partnerParams.get("job_id");
 
-            Utilities utilities = new Utilities(partner_id, api_key, sid_server, connectionTimeout, readTimeout);
-            response = utilities.get_job_status(user_id, job_id, options);
-        } catch (Exception e) {
-            throw e;
-        }
-
-        return response;
+        Utilities utilities = new Utilities(partner_id, api_key, sid_server, connectionTimeout, readTimeout);
+        return utilities.get_job_status(user_id, job_id, options);
+    }
+    
+    /***
+     *  Will query the backend for web session token with a specific timestamp
+     * @param timestamp the timestamp to generate the token from
+     * @param user_id
+     * @param job_id
+     * @param job_type
+     * @param product_type
+     * @param signature a previously generated signature
+     * @return TRUE or FALSE
+     */
+    @SuppressWarnings("unchecked")
+	public String get_hosted_web_session(Long timestamp, String user_id, String job_id, int job_type, String product_type, String signature) throws Exception {
```

Comment:
It doesn't make sense to require the developer calling this function to pass in a timestamp and signature. This should automatically generate the timestamp and signature and add it to the request, like we do for the submit_job method.

Reply:
Totally right!